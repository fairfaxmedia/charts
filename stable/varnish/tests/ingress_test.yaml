suite: Test ingress
templates:
  - ingress.yaml
tests:
  - it: has ingress by default
    asserts:
      - hasDocuments: { count: 1 }
      - matchSnapshot: {}
      - isKind:
          of: Ingress

  - it: matches the snapshot with one ingress
    values:
      - ./values/ok.yaml
    asserts:
      - hasDocuments: { count: 1 }
      - matchSnapshot: {}
      - isKind:
          of: Ingress
      - equal:
          path: metadata.name
          value: RELEASE-NAME-varnish-external
      - equal:
          path: spec.rules[0].http.paths[0].backend.serviceName
          value: RELEASE-NAME-varnish

  - it: uses correct service with service nameOverride applied
    values:
    - ./values/ok.yaml
    set:
      service:
        nameOverride: proxy
    asserts:
    - hasDocuments: { count: 1 }
    - matchSnapshot: {}
    - isKind:
        of: Ingress
    - equal:
        path: metadata.name
        value: RELEASE-NAME-varnish-external
    - equal:
        path: spec.rules[0].http.paths[0].backend.serviceName
        value: proxy
