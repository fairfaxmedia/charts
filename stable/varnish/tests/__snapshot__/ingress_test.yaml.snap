has ingress by default:
  1: |
    apiVersion: networking.k8s.io/v1beta1
    kind: Ingress
    metadata:
      labels:
        app: RELEASE-NAME-varnish
        chart: varnish-4.0.0
        component: proxy
        heritage: Tiller
        name: external
        release: RELEASE-NAME
      name: RELEASE-NAME-varnish-external
    spec:
      rules:
      - host: dummy-varnish.example.com
        http:
          paths:
          - backend:
              serviceName: RELEASE-NAME-varnish
              servicePort: 80
            path: /
matches the snapshot with one ingress:
  1: |
    apiVersion: networking.k8s.io/v1beta1
    kind: Ingress
    metadata:
      labels:
        app: RELEASE-NAME-varnish
        chart: varnish-4.0.0
        component: proxy
        heritage: Tiller
        name: external
        release: RELEASE-NAME
      name: RELEASE-NAME-varnish-external
    spec:
      rules:
      - host: dummy-varnish.example.com
        http:
          paths:
          - backend:
              serviceName: RELEASE-NAME-varnish
              servicePort: 80
            path: /
uses correct service with service nameOverride applied:
  1: |
    apiVersion: networking.k8s.io/v1beta1
    kind: Ingress
    metadata:
      labels:
        app: RELEASE-NAME-varnish
        chart: varnish-4.0.0
        component: proxy
        heritage: Tiller
        name: external
        release: RELEASE-NAME
      name: RELEASE-NAME-varnish-external
    spec:
      rules:
      - host: dummy-varnish.example.com
        http:
          paths:
          - backend:
              serviceName: proxy
              servicePort: 80
            path: /
